hotspot/src/main/java/io/deephaven/hotspot/HotSpotImpl.java

```diff
@@ -0,0 +1,19 @@
+package io.deephaven.hotspot;
+
+import com.google.auto.service.AutoService;
+import io.deephaven.util.hotspot.HotSpot;
+import sun.management.ManagementFactoryHelper;
+
+@AutoService(HotSpot.class)
+public class HotSpotImpl implements HotSpot {
+
+    @Override
+    public long getSafepointCount() {
+        return ManagementFactoryHelper.getVMManagement().getSafepointCount();
```

Comment:
I spent a little bit of time looking at the (internal, Sun-specific) APIs here, and I think we should do this slightly differently. See:
`sun.management.ManagementFactoryHelper#getHotspotRuntimeMBean`
It's labeled "This method is for testing only."
That said, it produces an MBean with an actual documented interface (`sun.management.HotspotRuntimeMBean`). It delegates internally to the same object, but using this class gets us on board with actual code samples on the Internet (I can't find any that are directly using `VMManagement`).

I assume we have adequate guards to make sure we don't try to do this on non-Hotspot VMs, and that @devinrsmith is looking at that.